src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -0,0 +1,153 @@
+      // Check left pixel
+      if (isStillMask) {
+        visit.push(leftPixelCoord);
+        visitSize += 1;
+      }
+      isStillMask = this.getIsStillMask(originalPixel, imgData, rightPixelCoord, tolerance, visited);
+      // Check right pixel
+      if (isStillMask) {
+        visit.push(rightPixelCoord);
+        visitSize += 1;
+      }
+      isStillMask = this.getIsStillMask(originalPixel, imgData, upPixelCoord, tolerance, visited);
+      // Check up pixel
+      if (isStillMask) {
+        visit.push(upPixelCoord);
+        visitSize += 1;
+      }
+      isStillMask = this.getIsStillMask(originalPixel, imgData, downPixelCoord, tolerance, visited);
+      // Check down pixel
+      if (isStillMask) {
+        visit.push(downPixelCoord);
+        visitSize += 1;
+      }
+    }  // end of while loop
+
+    return mask;
+  }
+
+  /** Judge current pixel's RGBA against original pixel's RGBA to see if it
+   *  can still be part of the mask (using tolerance criteria)
+   */
+  getIsStillMask(originalPixel: Array<number>, imgData: ImageData,
+                         pixelCoord: Array<number>, tolerance: number,
+                         visited: Set<number>): boolean {
+    let isStillMask: boolean = true;
+
+    // Preface; check if pixel is valid (indexing errs and repeat values)
+    let isValid: boolean = this.getIsValid(imgData.width, imgData.height, pixelCoord, visited);
+    if (!isValid) {
+      // Automatically not in mask b/c failed vailidity test
+      return !isStillMask;
+    }
+
+    // Get array of attributes of current pixel
+    let curX: number = pixelCoord[0];
+    let curY: number = pixelCoord[1];
+    let curPixel: Array<number> = this.dataArrayToRGBA(imgData, curX, curY);
+
+    // All attributes of the pixel (R,G,B, and A) must be within tolerance level
+    for (let i = 0; i < 4; i++) {
+      let upperTolerance: boolean = curPixel[i] > originalPixel[i] + tolerance;
+      let lowerTolerance: boolean = curPixel[i] < originalPixel[i] - tolerance;
+      let failsTolerance: boolean = upperTolerance || lowerTolerance;
+      if (failsTolerance) {
+        return !isStillMask;
+      }
+    }
+
+    return isStillMask;
+  }
+
+  // Check if @pixelCoord is in bounds and makes sure it's not a repeat coord
+  getIsValid(imgWidth: number, imgHeight: number, pixelCoord: Array<number>, visited: Set<number>): boolean {
+    let curX: number = pixelCoord[0];
+    let curY: number = pixelCoord[1];
+
+    let isValid: boolean = true;
+    // Check bounds of indexing
+    let yOutOfBounds: boolean = curY < 0 || curY > imgHeight - 1;
+    let xOutOfBounds: boolean = curX < 0 || curX > imgWidth - 1;
+    if (yOutOfBounds || xOutOfBounds) {
+      return !isValid;
+    }
+    // Do not push repeat coords to heap
+    let indexAsDataArray: number = this.coordToDataArrayIndices(curX, curY, imgWidth)[0];
+    if (visited.has(indexAsDataArray)) {
+      return !isValid;
+    }
+    visited.add(indexAsDataArray);
```

Comment:
I didn't expect this function to also add it to the visited array. I would move that out.

You want to make sure functions have a clear and distinct purpose.  It seems like the purpose of this function checks whether the pixel is valid so adding it to the visited array seems like a tangential side affect.